docs/source/using-discovery/bash.rst

```diff
@@ -0,0 +1,867 @@
+
+Bash commands perform various tasks within the shell environment. Commands span from basic functionalities (e.g., ``ls``, ``cd``, ``cp``, ``mv``, and ``rm``) to more advanced ones(e.g., ``grep`` and ``awk``). We cover these commands and more in this tutorial. Bash can also be used in scripts, allowing users to automate tasks and perform more complex operations via loops, conditional logic, and defining functions, which we cover at the end of this page.
+
+In summary, shell commands perform various tasks with the terminal.
+
+
+.. figure:: /images/terminal-view.png
+   :class: with-border
+   :width: 300
+   :alt: Terminal View.
+   :align: center
+
+   Terminal View
+
+Terminal
+=========
+The terminal - aka the command line interface (CLI) - is a text-based interface for interacting with an operating system. It is a way for users to interact with the system and perform tasks by typing commands and receiving text-based output.
+
+In contrast to graphical user interfaces (GUIs), the terminal provides a more direct and powerful way to interact with the system. Tasks that may require several steps in a GUI can often be accomplished much more quickly and efficiently in the terminal.
+
+Whether you are a beginner or an advanced user, the terminal provides a powerful and versatile interface for interacting with your operating system. With some theory and practice, you can use the terminal to accomplish a wide range of tasks and take control of your system in new and powerful ways.
+
+There are various terminal options (i.e., flavors) offered for different operating systems. `Power Shell`_ is available for Windows, Linux, and MacOS.
+
+Let us explore options and specifics for each operating system; Mac, Linux, and Windows terminals.
+
+MacOS
+------
+Mac OS comes with a default terminal program, but there are more advanced terminals available; `iTerm2`_ is one of the more popular choices.
+
+To launch the terminal:
+
+#. Press Command(⌘) + Space on your Mac keyboard (alternatively, press F4)
+#. Type in “Terminal”
+#. When you see Terminal in the Spotlight search list, click it to open the app.
+
+iTerm2 can be installed via the terminal using `Homebrew`_::
+
+   brew install --cask iterm2
+
+If Homebrew is not already installed, run the following command in the terminal before installing iTerm2::
+
+   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+Linux
+------
+Linux also comes with a default terminal program, but there are more advanced terminals available; `Terminator`_ is a popular choice.
+
+To download Terminator, open a terminal (``Ctrl+Alt+T`` is the shortcut to do so). Next, execute the following::
+
+   sudo add-apt-repository ppa:gnome-terminator
+   sudo apt-get update
+   sudo apt-get install terminator
```

Comment:
This is for Debian based OSs so you might want to point that out for people who are not using a Debian based OS. For example, there is yum install for Fedora.